pkg/oam/util/helper.go

```diff
@@ -302,18 +302,23 @@ func GetDefinition(ctx context.Context, cli client.Reader, definition client.Obj
 	appNs := GetDefinitionNamespaceWithCtx(ctx)
 	if err := cli.Get(ctx, types.NamespacedName{Name: definitionName, Namespace: appNs}, definition); err != nil {
 		if apierrors.IsNotFound(err) {
-			if err = cli.Get(ctx, types.NamespacedName{Name: definitionName, Namespace: oam.SystemDefinitionNamespace}, definition); err != nil {
-				if apierrors.IsNotFound(err) {
-					// compatibility code for old clusters those definition crd is cluster scope
-					var newErr error
-					if newErr = cli.Get(ctx, types.NamespacedName{Name: definitionName}, definition); checkRequestNamespaceError(newErr) {
-						return err
-					}
-					return newErr
-				}
+			return GetDefinitionFromDefaultNamespace(ctx, cli, definition, definitionName)
+		}
+		return err
+	}
+	return nil
+}
+
+// GetDefinitionFromDefaultNamespace get definition from "vela-system" namespace.
+func GetDefinitionFromDefaultNamespace(ctx context.Context, cli client.Reader, definition client.Object, definitionName string) error {
```

Comment:
Refactor with GetDefinitionFromNamespace(ctx context.Context, cli client.Reader, definition client.Object, definitionName string, namespace string)

Use it to get definition from appNs, then fallback to oam.SystemDefintionNamespace